### ISSUE TITLE
Will there a way to be able to choose a specific instruction set?

### ISSUE THREAD

**Issue by 1480c1** [OpenVisualCloud/SVT-AV1#165]
Is there a way to make it or to be able to choose instruction set at either run time or build time?
Main reason why is that if coveralls is actually going to be implemented, the code coverage will probably not rise much higherthan 30% <https://coveralls.io/github/1480c1/SVT-AV1?branch=coveralls> because when testing, it will only test with either the highest supported set or the lowest, and thus will only cover one set.

Tl:dr Can there be something like `-asm_type sse2` or `-asm_type ssse3` at runtime to choose an assembly instruction in between lowest assembly instruction and highest assembly instruction?

**Comment by EwoutH** [comment]
dav1d has a feature like that, could be very useful indeed

https://code.videolan.org/videolan/dav1d/merge_requests/476

**Comment by kylophone** [comment]
> dav1d has a feature like that, could be very useful indeed
> 
> https://code.videolan.org/videolan/dav1d/merge_requests/476

Yes, would be useful for performance testing.

**Comment by EwoutH** [comment]
I just did some perf reports (https://github.com/OpenVisualCloud/SVT-AV1/issues/98#issuecomment-494560666), a cpumask would be extremely useful to determine the differences between C, SSE, AVX2 and AVX512.

**Comment by hassount** [comment]
we welcome contributions here

**Comment by EwoutH** [comment]
Part of #262.

How is AVX-512 going to be split up? dav1d is going to use two cpumask options:

- `avx512`: Containing AVX-512 F, CD, ER, PF, VL, DQ, BW
- `avx512icl`: Containing AVX-512 F, CD, ER, PF, VL, DQ, BW, VPOPCNTDQ, IFMA, VBMI, VNNI, BF16, VBMI2, BITALG, VPCLMULQDQ, GFNI and VAES

**Comment by hassount** [comment]
sure, this sounds reasonable, we can adopt that

**Comment by kylophone** [comment]
I'm not super familiar with how this is implemented, but I'm curious. I'll take a look.
